[Inductor] added aten.cauchy_ decomp

min-jean-cho opened the pull request:
Fixes #91675

TODO: compare perf of decomposed tan --vs-- libdevice tan, aten tan for triton, cpp backeneds


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

min-jean-cho committed:
added cauchy decomp

min-jean-cho committed:
lint

jgong5 commented on torch/_decomp/decompositions.py:
```diff
@@ -1903,6 +1903,13 @@ def uniform(
+@register_decomposition(aten.cauchy_)
+def cauchy_(self, median=0, gamma=1, generator=None):
+    assert generator is None
+    M_PI = 3.141592653589793238462643383279502
```
why not using `math.pi`?

min-jean-cho replied:
Done, we should use `math.pi`

min-jean-cho committed:
(1) math.pi (2) move decomp to outplace wrapper

min-jean-cho committed:
Merge branch 'master' into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
remove mistake

min-jean-cho committed:
Merge branch 'minjean/inductor_decompose_aten_cauchy' of https://github.com/min-jean-cho/pytorch into minjean/inductor_decompose_aten_cauchy

jgong5 reviewed (approved):
It has test covered?

lezcano commented:
Could you add the relevant OpInfo test?

min-jean-cho committed:
(1) add opinfo for cauchy (2) add dtype check in decomp

min-jean-cho committed:
(1) add opinfo for cauchy (2) add dtype check in decomp

min-jean-cho committed:
Merge branch 'master' into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
Merge branch 'master' into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
merge conflict

min-jean-cho committed:
merge conflict

min-jean-cho committed:
merge conflict

ngimel commented:
Let's split this into first adding `tan` lowering to inductor, and then add cauchy decomposition that'll also require some changes to vmap testing. For inductot `tan` lowering you'd need to use `libdevice` for triton, see other math functions for examples of how this is done.

min-jean-cho commented:
> Let's split this into first adding `tan` lowering to inductor, and then add cauchy decomposition that'll also require some changes to vmap testing. For inductot `tan` lowering you'd need to use `libdevice` for triton, see other math functions for examples of how this is done.

Sounds good @ngimel, done in #92837

min-jean-cho committed:
cauchy opinfo, move out tan lowering

lezcano commented:
I'm not sure what @ngimel meant by "adding the cauchy decomp will require some changes to vmap", so let's wait to hear back from her on that.

min-jean-cho commented:
Thanks @lezcano, I saw earlier CI had some OpInfo `vmap` related failures to calling random operator

ngimel commented:
I was referring to a large number of vmap test failures

min-jean-cho committed:
cauchy opinfo

jansel commented on test/inductor/test_torchinductor_opinfo.py:
```diff
@@ -194,6 +194,7 @@ def process(device_type):
     "bernoulli": {f32, f64},
     "bincount": {i32, i64},
     "bucketize": {b8, f16, f32, f64, i32, i64},
+    "cauchy": {f16, f64},
```
Why does this add expected failures?

ngimel replied:
So what's the story with inductor opinfo testing now? It can't be passing because results for random sampling would be different between eager and inductor. Is the test not run?

min-jean-cho replied:
> It can't be passing because results for random sampling would be different between eager and inductor. Is the test not run?

Yes, the tests will be skipped because eager --vs-- inductor randomness are not comparable. (The tests are not skipped due to graph break of inplace ops; graph break has been resolved.)
But I don't know why some inductor randomness tests (eg: uniform cpu float32, float64) have passed. https://github.com/pytorch/pytorch/pull/93243/files#r1094149843

I'm not sure how generator is partially supported in inductor.

min-jean-cho committed:
cacuhy userdefinedobjectvariable remove

min-jean-cho committed:
after #93314

min-jean-cho committed:
Merge branch 'master' into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
lint

min-jean-cho committed:
Merge branch 'minjean/inductor_decompose_aten_cauchy' of https://github.com/min-jean-cho/pytorch into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
fix ci

min-jean-cho committed:
fix ci

min-jean-cho committed:
see cuda ci

min-jean-cho committed:
Merge branch 'pytorch:master' into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
fix ci

min-jean-cho committed:
Merge branch 'minjean/inductor_decompose_aten_cauchy' of https://github.com/min-jean-cho/pytorch into minjean/inductor_decompose_aten_cauchy

min-jean-cho committed:
fix ci

min-jean-cho commented:
Hi all, thank you for the review. I see all tests have passed now - I've made few minor changes since the last approvals, could you help have a final scan ?

min-jean-cho commented:
@pytorchbot merge
